test/test.js

```diff
@@ -0,0 +1,62 @@
+const fs = require('fs')
+const path = require('path')
+const postcss = require('postcss')
+const processor = require('../src')
+const { generateScopedName } = processor
+
+function generateInvalidCSS(css) {
+  css.walkDecls(function(decl) {
+    decl.value = decl.value.replace(/_colon_/g, ':') // because using a : in the tests would make it invalid CSS.
+  })
+}
+
+function normalize(str) {
+  return str.replace(/\r\n?/g, '\n').replace(/\n$/, '')
+}
+
+var testDir = path.join(__dirname, 'test-cases')
+fs.readdirSync(testDir).forEach(function(testCase) {
+  if (fs.existsSync(path.join(testDir, testCase, 'source.css'))) {
+    it('should ' + testCase.replace(/-/g, ' '), function() {
+      var input = normalize(
+        fs.readFileSync(path.join(testDir, testCase, 'source.css'), 'utf-8')
+      )
+      var expected, expectedError
+      if (fs.existsSync(path.join(testDir, testCase, 'expected.error.txt'))) {
+        expectedError = normalize(
+          fs.readFileSync(
+            path.join(testDir, testCase, 'expected.error.txt'),
+            'utf-8'
+          )
+        ).split('\n')[0]
+      } else {
+        expected = normalize(
```

Comment:
I think in future it's better to use snapshots here. It provides better string diffs as output and it's easier to find out what was wrong :)

I ported it to resolve-imports: https://github.com/css-modules/postcss-modules-resolve-imports/pull/21 and can help you here also.